rtengine/colortemp.cc

```diff
@@ -2963,6 +2963,13 @@ void ColorTemp::temp2mulxyz (double temp, const std::string &method, double &Xxy
 
     // We first test for specially handled methods
     const auto iterator = spectMap.find(method);
+    if (settings->observer10 == true) {
+        for (int i = 0; i < 97; i++) {
+            cie_colour_match_jd2[i][0] = cie_colour_match_jd[i][0];
+            cie_colour_match_jd2[i][1] = cie_colour_match_jd[i][1];;
+            cie_colour_match_jd2[i][2] = cie_colour_match_jd[i][2];
+        }
+    }
```

Comment:
If a user changes back to 2°, this makes it "stuck" on 10° until after an application restart. It would be better to pass a pointer or reference of the correct `cie_colour_match` to the functions that use them. This has the added benefits of avoiding these nearly 300 assignments and improving the code.

Reply:
@Desmis I would do something like this: [patch.diff.txt](https://github.com/Beep6581/RawTherapee/files/10255327/patch.diff.txt). You can apply this patch with `git apply patch.diff.txt`. This way, the behavior of the static functions only depend on the arguments.

Reply:
@Lawrence37 
Thank you. I apply this patch. works well... Code clean and fast.

Now I will look to include "settings" in main.

Jacques